src/App.tsx

```diff
@@ -1,24 +1,185 @@
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterTypeEnum';
 
-const USER_ID = 0;
+const USER_ID = 10592;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
-
-  return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
-  );
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [removeTodoIsClicked, setRemoveTodoIsClicked] = useState(false);
+  const [editTodoIsClicked, setEditTodoIsClicked] = useState(false);
+  const [filterMode, setFilterMode] = useState<FilterType>(FilterType.All);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [fetchTodosStatus, setFetchTodosStatus] = useState(false);
+  const [onEmptyFormSubmit, setOnEmptyFormSubmit] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [toggleTodoIsFailed, setToggleTodoIsFailed] = useState(false);
+  const [updatingTogosToggle, setUpdatingTodosToggle] = useState<number[]>([]);
+  const [deletingCompletedTodos,
+    setDeletingCompletedTodos] = useState<number[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const allTodosCompleted = useMemo(() => {
+    return filteredTodos.every((todo) => todo.completed);
+  }, [filteredTodos]);
+
+  const notAllTodosCompleted = useMemo(() => {
+    return filteredTodos.every((todo) => todo.completed);
+  }, [filteredTodos]);
+
+  const showNotification = removeTodoIsClicked
+  || editTodoIsClicked
+  || onEmptyFormSubmit
+  || toggleTodoIsFailed;
+
+  const getTodos = async () => {
+    try {
+      const data = await client.get(`/todos?userId=${USER_ID}`) as Todo[];
+
+      setTodos(data);
+      setFetchTodosStatus(true);
+    } catch {
+      setFetchTodosStatus(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodos();
+  }, []);
+
+  const filteredTodosArr = useMemo(() => {
+    return todos.filter((todo) => {
+      if (filterMode === FilterType.Active) {
+        return !todo.completed;
+      }
+
+      if (filterMode === FilterType.Completed) {
+        return todo.completed;
+      }
+
+      return todo;
+    });
+  }, [todos, filterMode]);
+
+  useEffect(() => {
+    setFilteredTodos(filteredTodosArr);
+  }, [filteredTodosArr]);
+
+  const handleToggleAllClick = async () => {
+    const updatedTodos = filteredTodosArr.map((todo) => ({
+      ...todo,
+      completed: !allTodosCompleted,
+    }));
+
+    try {
+      setUpdatingTodosToggle(updatedTodos.map((todo) => todo.id));
+
+      await Promise.all(
+        updatedTodos.map(async (todo) => {
+          await fetch(`https://mate.academy/students-api/todos/${todo.id}`,
```

Comment:
Using api calls like this is not a good practice. Pls, move all API calls to a separate file or even better to create a folder `api` and put all api calls there in separate consts like `getTodos`, `updateTodos` etc.